### PR TITLE
Replace SDSF Job Status with `zowex`

### PR DESCRIPTION
**Pull request by MarkAckert**
This PR replaces SDSF usage in the `zos-jes.ts` module with `zowex`, which uses SSI function code 80 to obtain jobs information.

Notable changes:
*  [Zowe Native Proto Server Side](https://github.com/zowe/zowe-native-proto/#architecture) is part of the PAX build and placed in `bin/utils`. `zowex` is copied over, while `zoweax` and `zowed` are not (for now). They may be used in the future.
* Job completion text is not available in the `zowex job vs` output today and is removed. This was mostly a footnote on some job submission logging.
* As a byproduct of this change, JES3 should work since it also supports SSI 80.

Todo:
* ~~Validation of PAX build~~ Includes `zowex`, CICD test failures are the known APIML ones in staging
* ~~Testing / Snapshot updates~~ No updates, log changes were in debug logs which aren't captured in tests. All `ZWE` tests passing.
* ~~Better to delete `zowed` and `zoweax`?~~ Not copied over
* ~~Edge-case test: what if a submitted job is cancelled and purged before `zowex job vs` is run?~~ Done. Marked as a FLAKY test so it won't run with CI/CD, but the snapshot is there and the behavior was correct.
* ~~CLI Squad will produce a formal release of zowe-native-proto for us. I copied a nightly build for this PR into a temporary folder on Artifactory.~~ Done

### CODE REVIEW

File: bin/libs/zos-jes.ts [hunk 1]
```diff
@@ -98,95 +98,96 @@ export function submitJob(jclFileOrContent: string, printJobDebug:boolean=true,
...
 }
 
-export function waitForJob(jobid: string): {jobcctext?: string, jobcccode?: string, jobid?: string, jobname?: string, rc: number} {
-  let jobstatus;
-  let jobname;
-  let jobcctext;
-  let jobcccode;    
-  let is_jes3;
+/**
+ * Returns job name, completion code, and status. If we cannot retrieve the completion code, it is set to -1.
```
Comment (Martin-Zeithaml) [inline bin/libs/zos-jes.ts]: `If we cannot retrieve the completion code, it is set to -1. `
The default compCode is empty string (L113).

File: bin/libs/zos-jes.ts [hunk 2]
```diff
@@ -98,95 +98,96 @@ export function submitJob(jclFileOrContent: string, printJobDebug:boolean=true,
...
+      JOB05609,CANCELED,LONGJOB,OUTPUT,J0005609SVSCJES2E131ED16.......:
+      JOB05607,CANCELED,LONGJOB,OUTPUT,J0005607SVSCJES2E131ED14.......:
+      JOB05602,CC 0000,ATLJ0000,OUTPUT,J0005602SVSCJES2E131ED0F.......:
+      JOB05586,CC 0000,ATLJ0000,OUTPUT,J0005586SVSCJES2E131ECF4.......:
+      TSU05815,,USERABC,ACTIVE,T0005815SVSCJES2E132110A.......:
+      */
+      const columns = line.split(',').reverse(); //TODO: commas should be safe, not legal jobname/correlator? how about user id?
+      status = columns[1];
+      jobName = columns[2];
+      compCode = columns[3];
```
Comment (Martin-Zeithaml) [inline bin/libs/zos-jes.ts]: This will overwrite the default return values for incorrect input (empty jobId) or jobId, which was deleted:
* `jobId = ''` returns `{"status":"  --response-format-csv"}`
* `jobId = 'JOB12345' returns `{}`

We should process the `result.out` only if `result.rc == 0`.

File: bin/README.md [hunk 3]
```diff
@@ -54,10 +54,11 @@ Please be aware of using functions marked as `@experimental`. These functions ma
 
 `bin/utils` directory holds several utility tools used by Zowe, and you can take advantage of them too.
 
-- `bin/utils/opercmd.rex`: To issue operator command on z/OS. This script can only run on z/OS.
+- `bin/utils/opercmd.rex`: To issue operator command on z/OS. This script can only run on z/OS and requires SDSF.
 - `bin/utils/curl.js`: This node.js script works similar to popular Linux tool `curl`. It can make HTTP/HTTPS request and display response.
-- `bin/utils/getesm`: Executable to get the name of External Security Manager
+- `bin/utils/getesm`: Executable to get the name of External Security Manager.
 - `bin/utils/certificate-analyser.jar`: Java based tool to verify the certificates. See [README](https://github.com/zowe/api-layer/blob/v3.x.x/certificate-analyser/README.md) for more information.
+- `bin/utils/zowex`: Zowe Native Protocol CLI for various tasks (console, jobs, datasets, ...), use `zowex --help` for more information.
```
Comment (Martin-Zeithaml) [inline bin/README.md]: We can mark it as `@experimental`: it is used currently only internally and it is not expected to be used directly by customers.

**Comment by Martin-Zeithaml** [comment]
## Quick testing
```
[1/4][expected rc=  0, result rc=  0  ]: /4406/bin/zwe install -c ./yaml/zowe.basic.yaml --allow-overwrite -vv
[2/4][expected rc=  0, result rc=  0  ]: /4406/bin/zwe init generate -c ./yaml/zowe.basic.yaml -vv
[3/4][expected rc=  0, result rc=  0  ]: /4406/bin/zwe init mvs -c ./yaml/zowe.basic.yaml --generate -vv
[4/4][expected rc=  0, result rc=  0  ]: /4406/bin/zwe init stc -c ./yaml/zowe.basic.yaml --generate -vv
```
## Example of "new" waiting

```
  * Succeeded with job ID JOB07937
  * Exit code: 0
  * Output:
    JOB JOB07937 submitted from stdin
- Wait for job JOB07937 completed, starting at Thu Jul 17 2025 11:58:08 GMT+0000.
  * Wait for 1 seconds
-- Running /4406/bin/utils/zowex job vs JOB07937 --rfc
  * Job (ZWEGENER) status is ACTIVE,RC=
  * Wait for 5 seconds
-- Running /4406/bin/utils/zowex job vs JOB07937 --rfc
  * Job (ZWEGENER) status is ACTIVE,RC=
  * Wait for 10 seconds
-- Running /4406/bin/utils/zowex job vs JOB07937 --rfc
  * Job (ZWEGENER) status is OUTPUT,RC=CC 0000
  * Job status check done at Thu Jul 17 2025 11:58:25 GMT+0000.
  * Job (ZWEGENER) exits with code CC 0000.
Job ZWEGENER(JOB07937) completed with RC=0
```

File: bin/libs/zos-jes.ts (hunk 2 above)
Comment (MarkAckert) [reply bin/libs/zos-jes.ts]: Confirmed `zowex` rc>0 for those error conditions and updated the outer block

File: bin/libs/zos-jes.ts (hunk 1 above)
Comment (MarkAckert) [reply bin/libs/zos-jes.ts]: Updated to match current behavior.

**Comment by MarkAckert** [comment]
This is updated and ready for review @Martin-Zeithaml @1000TurquoisePogs 

One update to the PR description - the JES3 support *should* work but isn't tested. If someone has a JES3 system and can test that'd be appreciated.

**Review by Martin-Zeithaml — APPROVED** [review]
